### PR TITLE
feat(anthropic): Emit AI Client Spans for synchronous `messages.stream()`

### PR DESCRIPTION
**Pull request by alexander-alderman-webb**
### Description

Patch `Messages.stream()` and `MessageStreamManager.__enter__()` to create AI Client Spans.

Re-use existing code for setting attributes on AI Client Spans based on function arguments, and adapt tests that return a synchronous response stream with `create(stream=True)`.

#### Reminders
- Please add tests to validate your changes, and lint your code using `tox -e linters`.
- Add GH Issue ID _&_ Linear ID (if applicable)
- PR title should use [conventional commit](https://develop.sentry.dev/engineering-practices/commit-messages/#type) style (`feat:`, `fix:`, `ref:`, `meta:`)
- For external contributors: [CONTRIBUTING.md](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md), [Sentry SDK development docs](https://develop.sentry.dev/sdk/), [Discord community](https://discord.gg/Ww9hbqr)

### CODE REVIEW

File: sentry_sdk/integrations/anthropic.py [hunk 1]
```diff
@@ -84,6 +91,11 @@ def setup_once() -> None:
         Messages.create = _wrap_message_create(Messages.create)
         AsyncMessages.create = _wrap_message_create_async(AsyncMessages.create)
 
+        Messages.stream = _wrap_message_stream(Messages.stream)
+        MessageStreamManager.__enter__ = _wrap_message_stream_manager_enter(
+            MessageStreamManager.__enter__
+        )
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py:98]: good one

[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py:98]: coming up

File: sentry_sdk/integrations/anthropic.py [hunk 2]
```diff
@@ -253,27 +265,32 @@ def _transform_system_instructions(
...
 ) -> None:
     """
     Set input data for the span based on the provided keyword arguments for the anthropic message creation.
     """
     span.set_data(SPANDATA.GEN_AI_OPERATION_NAME, "chat")
-    system_instructions: "Union[str, Iterable[TextBlockParam]]" = kwargs.get("system")  # type: ignore
-    messages = kwargs.get("messages")
     if (
         messages is not None
         and len(messages) > 0
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: out of scope since this is an existing problem

File: sentry_sdk/integrations/anthropic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py:98]: added a check https://github.com/getsentry/sentry-python/pull/5565/changes/bf009ca848482be62f75f97d119a8d8ff5dd46b9

File: sentry_sdk/integrations/anthropic.py [hunk 3]
```diff
@@ -664,6 +723,104 @@ async def _sentry_patched_create_async(*args: "Any", **kwargs: "Any") -> "Any":
...
+
+    if model is None:
+        model = ""
+
+    span = get_start_span_function()(
+        op=OP.GEN_AI_CHAT,
+        name=f"chat {model}".strip(),
+        origin=AnthropicIntegration.origin,
+    )
+    span.__enter__()
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: will need to check this edge case

File: tests/integrations/anthropic/test_anthropic.py [hunk 4]
```diff
@@ -2407,6 +2800,72 @@ def test_input_tokens_include_cache_read_streaming(sentry_init, capture_events):
 
     (span,) = events[0]["spans"]
 
+    # input_tokens should be total: 19 + 2846 = test_stream_messages_input_tokens_include_cache_read_streaming
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply tests/integrations/anthropic/test_anthropic.py:2919]: this is copy pasted from another test.

File: sentry_sdk/integrations/anthropic.py [hunk 5]
```diff
@@ -329,32 +348,74 @@ def _set_input_data(
...
+        integration=integration,
+        max_tokens=max_tokens,
+        messages=messages,
+        model=model,
+        system=system,
+        temperature=temperature,
+        top_k=top_k,
+        top_p=top_p,
+        tools=tools,
+    )
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: Nice I was missing an attribute: https://github.com/getsentry/sentry-python/pull/5565/changes/8e4b6e893cdad36780a2e156a2069bb163793159

File: sentry_sdk/integrations/anthropic.py [hunk 6]
```diff
@@ -674,6 +737,90 @@ async def _sentry_patched_create_async(*args: "Any", **kwargs: "Any") -> "Any":
...
+            model = ""
+
+        span = get_start_span_function()(
+            op=OP.GEN_AI_CHAT,
+            name=f"chat {model}".strip(),
+            origin=AnthropicIntegration.origin,
+        )
+        span.__enter__()
+
+        span.set_data(SPANDATA.GEN_AI_RESPONSE_STREAMING, True)
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py:787]: my bad I'll fix this

[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py:787]: https://github.com/getsentry/sentry-python/pull/5565/commits/1e4fb121677e57a32a5d670a9bd5baef2940ce63

File: sentry_sdk/integrations/anthropic.py [hunk 7]
```diff
@@ -335,25 +354,164 @@ def _set_input_data(
...
+        (
+            model,
+            usage,
+            content_blocks,
+        ) = _collect_ai_data(
+            event,
+            model,
+            usage,
+            content_blocks,
+        )
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: was me resolving merge conflicts wrong, it's fixed now

File: sentry_sdk/integrations/anthropic.py [hunk 8]
```diff
@@ -695,6 +729,90 @@ async def _sentry_patched_create_async(*args: "Any", **kwargs: "Any") -> "Any":
...
+        span.__enter__()
+
+        span.set_data(SPANDATA.GEN_AI_RESPONSE_STREAMING, True)
+        _set_common_input_data(
+            span=span,
+            integration=integration,
+            max_tokens=self._max_tokens,
+            messages=self._messages,
+            model=model,
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: good one

[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/anthropic.py]: done https://github.com/getsentry/sentry-python/pull/5565/changes/f88357c64345bedbdbf991c12898e764229727a0
